--- pmx2506.for.orig	2005-09-27 23:38:14.000000000 +0900
+++ pmx2506.for	2005-09-27 23:36:43.000000000 +0900
@@ -624,8 +624,8 @@
         read(*,'(a)')jobname
         numargs = 1
       else 
-        call getarg(1,jobname,idum) ! May need to replace this w/ next line 
-c        call getarg(1,jobname) 
+c        call getarg(1,jobname,idum) ! May need to replace this w/ next line 
+        call getarg(1,jobname) 
       end if
 10    ljob = lenstr(jobname,44)
       if (ljob .gt. 44) then
@@ -637,8 +637,8 @@
       else if (numargs .eq. 2) then
         if (ljob.eq.2 .and. jobname(1:2).eq.'-o') then
           optimize = .true.
-          call getarg(2,jobname,idum) ! May need to replace this w/ next line 
-c          call getarg(2,jobname) 
+c          call getarg(2,jobname,idum) ! May need to replace this w/ next line 
+          call getarg(2,jobname) 
           numargs = 1 
           go to 10
         else
