diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5f26e71..b4c389a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ get_filename_component(CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" ABSOLUTE)
 # RPath
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 # Should we build with documentation
@@ -153,7 +153,7 @@ if(XDMF_WRAP_PYTHON)
 
     string(TOUPPER ${python_name} python_name_upper)
     set(${python_name_upper}_PYTHON
-      ${CMAKE_INSTALL_PREFIX}/lib/python/${python_name}.py)
+      ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/python/${python_name}.py)
     get_directory_property(${python_name}Parent PARENT_DIRECTORY)
     if(NOT "${${python_name}Parent}" STREQUAL "")
       set(${python_name_upper}_PYTHON ${${python_name_upper}_PYTHON}
@@ -161,9 +161,9 @@ if(XDMF_WRAP_PYTHON)
     endif(NOT "${${python_name}Parent}" STREQUAL "")
 
     install(FILES ${CMAKE_BINARY_DIR}/${python_name}.py
-      DESTINATION lib/python)
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/python)
     install(TARGETS ${SWIG_MODULE_${python_name}Python_REAL_NAME}
-      DESTINATION lib/python)
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/python)
   endmacro(XDMF_SWIG_PYTHON)
 endif(XDMF_WRAP_PYTHON)
 
@@ -210,7 +210,7 @@ if(XDMF_WRAP_JAVA)
       set(separator "\;")
     else(WIN32)
       set(java_files ${XDMF_JAVA_DIR}/*.java)
-      set(java_dest "lib/java")
+      set(java_dest "${CMAKE_INSTALL_LIBDIR}/java")
       set(separator ":")
     endif(WIN32)
 
@@ -331,7 +331,7 @@ if(NOT XDMF_BUILD_CORE_ONLY)
   endif(WIN32)
   
   if(UNIX)
-    set(XDMF_LIBRARY ${CMAKE_INSTALL_PREFIX}/lib/libXdmf.so)
+    set(XDMF_LIBRARY ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/libXdmf.so)
   endif(UNIX)
 
   if(APPLE)
@@ -347,9 +347,9 @@ if(NOT XDMF_BUILD_CORE_ONLY)
   )
   install(FILES ${XdmfHeaders} DESTINATION include)
   install(TARGETS Xdmf
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} 
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif(NOT XDMF_BUILD_CORE_ONLY)
 
 xdmf_create_config_file(${PROJECT_NAME})
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 934939e..f3e1c27 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -19,7 +19,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_BINARY_DIR})
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 option(XDMF_BUILD_DSM OFF)
@@ -153,7 +153,7 @@ if(WIN32)
 endif(WIN32)
 
 if(UNIX)
-    set(XDMFCORE_LIBRARY ${CMAKE_INSTALL_PREFIX}/lib/libXdmfCore.so
+    set(XDMFCORE_LIBRARY ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/libXdmfCore.so
         PARENT_SCOPE)
 endif(UNIX)
 
@@ -170,9 +170,9 @@ file(GLOB LokiHeaders loki/*.h)
 install(FILES ${XdmfCoreHeaders} DESTINATION include)
 install(FILES ${LokiHeaders} DESTINATION include/loki)
 install(TARGETS XdmfCore
-  RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib)
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} 
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 set(XdmfCore_INCLUDE_DIRS
   ${Boost_INCLUDE_DIRS}
@@ -183,9 +183,9 @@ set(XdmfCore_INCLUDE_DIRS
   CACHE INTERNAL "")
 
 set(XDMF_INCLUDE_DIRS ${CMAKE_INSTALL_PREFIX}/include PARENT_SCOPE)
-set(XDMF_LIBRARIES ${XDMF_LIBRARIES} ${CMAKE_INSTALL_PREFIX}/lib)
+set(XDMF_LIBRARIES ${XDMF_LIBRARIES} ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
 set(XDMF_DIR ${CMAKE_INSTALL_PREFIX} PARENT_SCOPE)
-set(XDMF_BINARIES ${XDMF_BINARIES} ${CMAKE_INSTALL_PREFIX}/bin)
+set(XDMF_BINARIES ${XDMF_BINARIES} ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR})
 
 if(XDMF_BUILD_DSM)
   add_subdirectory(dsm)
