diff -ru Pixie-old/src/ri/texture.cpp Pixie/src/ri/texture.cpp
--- Pixie-old/src/ri/texture.cpp	2008-03-26 00:43:37.000000000 +0900
+++ Pixie/src/ri/texture.cpp	2009-03-02 11:40:07.000000000 +0900
@@ -1351,6 +1351,9 @@
 								cP[COMP_Y]			=	(D0[COMP_Y]*(1 - x) + D1[COMP_Y]*x)*(1-y) + (D2[COMP_Y]*(1 - x) + D3[COMP_Y]*x)*y;
 								cP[COMP_Z]			=	(D0[COMP_Z]*(1 - x) + D1[COMP_Z]*x)*(1-y) + (D2[COMP_Z]*(1 - x) + D3[COMP_Z]*x)*y;
 								cP[3]				=	1;
+								if(isnan(cP[COMP_X]))break;
+								if(isnan(cP[COMP_Y]))break;
+								if(isnan(cP[COMP_Z]))break;
 
 								mulmp4(tmp,header.toNDC,cP);
 								s					=	tmp[0] / tmp[3];

