--- a/projects/unix/paraview.bundle.unix.cmake	2019-10-24 12:24:57.716418174 +0900
+++ b/projects/unix/paraview.bundle.unix.cmake	2019-10-24 12:28:03.437414251 +0900
@@ -1,5 +1,5 @@
 set(library_paths
-  "${superbuild_install_location}/lib")
+  "${superbuild_install_location}/lib64")
 
 if (Qt5_DIR)
   list(APPEND library_paths
@@ -23,7 +23,7 @@
 
 foreach (executable IN LISTS paraview_executables)
   superbuild_unix_install_program("${superbuild_install_location}/bin/${executable}"
-    "lib"
+    "lib64"
     SEARCH_DIRECTORIES  "${library_paths}"
     INCLUDE_REGEXES     ${include_regexes}
     EXCLUDE_REGEXES     ${exclude_regexes})
@@ -38,7 +38,7 @@
 
 foreach (paraview_plugin IN LISTS paraview_plugins)
   superbuild_unix_install_plugin("${paraview_plugin}.so"
-    "lib"
+    "lib64"
     "${paraview_plugin_path}/${paraview_plugin}"
     LOADER_PATHS    "${library_paths}"
     INCLUDE_REGEXES ${include_regexes}
@@ -62,15 +62,15 @@
 
   foreach (mesa_library IN LISTS mesa_libraries)
     file(GLOB lib_filenames
-      RELATIVE "${superbuild_install_location}/lib"
-      "${superbuild_install_location}/lib/lib${mesa_library}.so*")
+      RELATIVE "${superbuild_install_location}/lib64"
+      "${superbuild_install_location}/lib64/lib${mesa_library}.so*")
 
     foreach (lib_filename IN LISTS lib_filenames)
       superbuild_unix_install_plugin("${lib_filename}"
-        "lib${suffix}"
-        "lib"
+        "lib64${suffix}"
+        "lib64"
         LOADER_PATHS  "${library_paths}"
-        LOCATION      "lib${suffix}")
+        LOCATION      "lib64${suffix}")
     endforeach ()
   endforeach ()
 endif ()
@@ -83,15 +83,15 @@
 
   foreach (nvidiaindex_library IN LISTS nvidiaindex_libraries)
     file(GLOB lib_filenames
-      RELATIVE "${superbuild_install_location}/lib"
-      "${superbuild_install_location}/lib/lib${nvidiaindex_library}.so*")
+      RELATIVE "${superbuild_install_location}/lib64"
+      "${superbuild_install_location}/lib64/lib${nvidiaindex_library}.so*")
 
     foreach (lib_filename IN LISTS lib_filenames)
       superbuild_unix_install_plugin("${lib_filename}"
-        "lib"
-        "lib"
+        "lib64"
+        "lib64"
         LOADER_PATHS  "${library_paths}"
-        LOCATION      "lib"
+        LOCATION      "lib64"
         EXCLUDE_REGEXES ".*/libcuda.so.*")
     endforeach ()
   endforeach ()
@@ -103,15 +103,15 @@
 
   foreach (osprayextra_library IN LISTS osprayextra_libraries)
     file(GLOB lib_filenames
-      RELATIVE "${superbuild_install_location}/lib"
-      "${superbuild_install_location}/lib/lib${osprayextra_library}.so*")
+      RELATIVE "${superbuild_install_location}/lib64"
+      "${superbuild_install_location}/lib64/lib${osprayextra_library}.so*")
 
     foreach (lib_filename IN LISTS lib_filenames)
       superbuild_unix_install_plugin("${lib_filename}"
-        "lib"
-        "lib"
+        "lib64"
+        "lib64"
         LOADER_PATHS  "${library_paths}"
-        LOCATION      "lib")
+        LOCATION      "lib64")
     endforeach ()
   endforeach ()
 endif ()
@@ -125,15 +125,15 @@
 
   foreach (visrtxextra_library IN LISTS visrtxextra_libraries)
     file(GLOB lib_filenames
-      RELATIVE "${superbuild_install_location}/lib"
-      "${superbuild_install_location}/lib/${visrtxextra_library}.so*")
+      RELATIVE "${superbuild_install_location}/lib64"
+      "${superbuild_install_location}/lib64/${visrtxextra_library}.so*")
 
     foreach (lib_filename IN LISTS lib_filenames)
       superbuild_unix_install_plugin("${lib_filename}"
-        "lib"
-        "lib"
+        "lib64"
+        "lib64"
         LOADER_PATHS  "${library_paths}"
-        LOCATION      "lib"
+        LOCATION      "lib64"
         SEARCH_DIRECTORIES "/usr/lib64/libglvnd" "/usr/lib/libglvnd"
         EXCLUDE_REGEXES ".*/libGLX.so.*")
     endforeach ()
@@ -142,7 +142,7 @@
 
 if (python_enabled)
   file(GLOB egg_dirs
-    "${superbuild_install_location}/lib/python${superbuild_python_version}/site-packages/*.egg/")
+    "${superbuild_install_location}/lib64/python${superbuild_python_version}/site-packages/*.egg/")
   if (python2_built_by_superbuild)
     include(python2.functions)
     superbuild_install_superbuild_python2(
@@ -160,21 +160,21 @@
   #
   # Related issue: https://gitlab.kitware.com/paraview/paraview-superbuild/issues/120
   superbuild_unix_install_python(
-    LIBDIR              "lib"
+    LIBDIR              "lib64"
     MODULES             paraview
                         vtk
                         vtkmodules
                         ${python_modules}
     INCLUDE_REGEXES     ${include_regexes}
     EXCLUDE_REGEXES     ${exclude_regexes}
-    MODULE_DIRECTORIES  "${superbuild_install_location}/lib/python${superbuild_python_version}/site-packages"
+    MODULE_DIRECTORIES  "${superbuild_install_location}/lib64/python${superbuild_python_version}/site-packages"
                         ${egg_dirs}
     LOADER_PATHS        "${library_paths}")
 
   if (matplotlib_built_by_superbuild)
     install(
-      DIRECTORY   "${superbuild_install_location}/lib/python${superbuild_python_version}/site-packages/matplotlib/mpl-data/"
-      DESTINATION "lib/python${superbuild_python_version}/site-packages/matplotlib/mpl-data"
+      DIRECTORY   "${superbuild_install_location}/lib64/python${superbuild_python_version}/site-packages/matplotlib/mpl-data/"
+      DESTINATION "lib64/python${superbuild_python_version}/site-packages/matplotlib/mpl-data"
       COMPONENT   superbuild)
   endif ()
 endif ()
@@ -188,13 +188,13 @@
     hydra_persist)
   foreach (mpi_executable IN LISTS mpi_executables_standalone)
     superbuild_unix_install_plugin("${superbuild_install_location}/bin/${mpi_executable}"
-      "lib"
+      "lib64"
       "bin")
   endforeach ()
   foreach (mpi_executable IN LISTS mpi_executables_standalone mpi_executables_paraview)
     superbuild_unix_install_plugin("${superbuild_install_location}/bin/${mpi_executable}"
-      "lib"
-      "lib")
+      "lib64"
+      "lib64")
   endforeach ()
 endif ()
 
@@ -211,7 +211,7 @@
   get_filename_component(qt5_plugin_group "${qt5_plugin_group}" NAME)
 
   superbuild_unix_install_plugin("${qt5_plugin_path}"
-    "lib"
+    "lib64"
     "plugins/${qt5_plugin_group}/"
     LOADER_PATHS    "${library_paths}"
     INCLUDE_REGEXES ${include_regexes}
