diff -urN 2008_0918_RT2860_Linux_STA_v1.8.0.0.org/os/linux/config.mk 2008_0918_RT2860_Linux_STA_v1.8.0.0/os/linux/config.mk
--- 2008_0918_RT2860_Linux_STA_v1.8.0.0.org/os/linux/config.mk	2008-10-21 22:51:01.000000000 -0500
+++ 2008_0918_RT2860_Linux_STA_v1.8.0.0/os/linux/config.mk	2008-10-21 22:52:41.000000000 -0500
@@ -1,14 +1,14 @@
 # Support ATE function
-HAS_ATE=n
+HAS_ATE=y
 
 # Support 28xx QA ATE function
-HAS_28xx_QA=n
+HAS_28xx_QA=y
 
 # Support Wpa_Supplicant
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 # Support Native WpaSupplicant for Network Maganger
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n
